compiler-core/src/ast/typed.rs

```diff
@@ -1120,6 +1120,25 @@ impl TypedExpr {
             _ => false,
         }
     }
+
+    pub(crate) fn expression_flattening_block(&self) -> Option<&TypedExpr> {
```

Comment:
Document what this does please. 🙏 

The name doesn't seem to match what it does, and it seems very specific to `CollapseNestedCase`, so I'm not convinced that this belongs as a method on the generic `TypedExpr` class.